pkg/rbac/clusterScopedResources_test.go

```diff
@@ -0,0 +1,112 @@
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockPool := pgxpoolmock.NewMockPgxPool(ctrl)
+	return mockPool, Cache{
+		shared: sharedList{},
+		pool:   mockPool,
+	}
+}
+
+func Test_getResources_emptyCache(t *testing.T) {
+
+	mockpool, cache := newResourcesListCache(t)
+	columns := []string{"apigroup", "kind"}
+	pgxRows := pgxpoolmock.NewRows(columns).AddRow("Node", "addon.open-cluster-management.io").ToPgxRows()
+
+	mockpool.EXPECT().Query(gomock.Any(),
+		gomock.Eq(`SELECT DISTINCT COALESCE("data"->>'apigroup', '') AS "apigroup", COALESCE("data"->>'kind', '') AS "kind" FROM "search"."resources" WHERE ("cluster"::TEXT = 'local-cluster' AND ("data"->>'namespace' IS NULL))`),
+		gomock.Eq([]interface{}{}),
+	).Return(pgxRows, nil)
+
+	result, err := cache.checkUserResources()
+
+	if len(result.resources) != 0 {
+		t.Error("Resources not in cache.")
+	}
+	if err != nil {
+		t.Error("Unexpected error while obtaining cluster-scoped resources.", err)
+	}
+
+}
+
+func Test_getResouces_usingCache(t *testing.T) {
+
+	mockpool, cache := newResourcesListCache(t)
+	columns := []string{"apigroup", "kind"}
+	pgxRows := pgxpoolmock.NewRows(columns).AddRow("Node", "addon.open-cluster-management.io").ToPgxRows()
+
+	mockpool.EXPECT().Query(gomock.Any(),
+		gomock.Eq(`SELECT DISTINCT COALESCE("data"->>'apigroup', '') AS "apigroup", COALESCE("data"->>'kind', '') AS "kind" FROM "search"."resources" WHERE ("cluster"::TEXT = 'local-cluster' AND ("data"->>'namespace' IS NULL))`),
+		gomock.Eq([]interface{}{}),
+	).Return(pgxRows, nil)
+
+	resourcemap := make(map[string][]string)
+	var apigroups string
+	var kinds []string
+
+	kinds = append(kinds, "kind1", "kind2")
+	apigroups = "apigroup1"
+
+	resourcemap[apigroups] = kinds
+	cache.shared = sharedList{
+		updatedAt: time.Now(),
+		resources: resourcemap,
+	}
+
+	result, err := cache.checkUserResources()
+
+	if len(result.resources) == 0 {
+		t.Error("Expected resources in cache.")
+	}
+
+	if err != nil {
+		t.Error("Unexpected error while obtaining cluster-scoped resources.", err)
+	}
+}
+
+func Test_getResources_expiredCache(t *testing.T) {
+
+	mockpool, cache := newResourcesListCache(t)
+	columns := []string{"apigroup", "kind"}
+	pgxRows := pgxpoolmock.NewRows(columns).AddRow("Node", "addon.open-cluster-management.io").ToPgxRows()
+
+	mockpool.EXPECT().Query(gomock.Any(),
+		gomock.Eq(`SELECT DISTINCT COALESCE("data"->>'apigroup', '') AS "apigroup", COALESCE("data"->>'kind', '') AS "kind" FROM "search"."resources" WHERE ("cluster"::TEXT = 'local-cluster' AND ("data"->>'namespace' IS NULL))`),
+		gomock.Eq([]interface{}{}),
+	).Return(pgxRows, nil)
+
+	resourcemap := make(map[string][]string)
+	var apigroups string
+	var kinds []string
+
+	kinds = append(kinds, "kind1", "kind2")
+	apigroups = "apigroup1"
+
+	resourcemap[apigroups] = kinds
+	cache.shared = sharedList{
+		updatedAt: time.Now().Add(time.Duration(-2) * time.Minute),
```

Comment:
I'd set this to a bigger difference like 3 mins to make it clear that it's expired and avoid possible test flakiness if everything runs within the same time unit.

Reply:
Done.